### ISSUE TITLE
WARNING NaN or Inf found in input tensor

### ISSUE THREAD

**Issue by apoca909** [wenet-e2e/wenet#496]
**Describe the bug**
2021-07-07 04:50:33,734 WARNING NaN or Inf found in input tensor.
2021-07-07 04:50:34,970 WARNING NaN or Inf found in input tensor.
2021-07-07 04:50:36,199 WARNING NaN or Inf found in input tensor.
2021-07-07 04:50:37,422 WARNING NaN or Inf found in input tensor.
2021-07-07 04:50:38,644 WARNING NaN or Inf found in input tensor.
2021-07-07 04:50:39,836 WARNING NaN or Inf found in input tensor.
2021-07-07 04:50:40,966 WARNING NaN or Inf found in input tensor.
2021-07-07 04:50:41,064 DEBUG TRAIN Batch 20100/282164 loss nan loss_att nan loss_ctc nan lr 0.00004021 rank 3
2021-07-07 04:50:41,080 DEBUG TRAIN Batch 20100/282164 loss nan loss_att nan loss_ctc nan lr 0.00004021 rank 0
2021-07-07 04:50:41,082 DEBUG TRAIN Batch 20100/282164 loss nan loss_att nan loss_ctc nan lr 0.00004021 rank 2
2021-07-07 04:50:41,084 DEBUG TRAIN Batch 20100/282164 loss nan loss_att nan loss_ctc nan lr 0.00004021 rank 1

someone has any ideas?

**Comment by apoca909** [comment]
the  audio training data has something wrong.

**Comment by apoca909** [comment]
the encoder subsampling_class  self.conv[0].weight as follow :

p self.conv[0].weight
Parameter containing:
tensor([[[[nan, nan, nan],
          [nan, nan, nan],
          [nan, nan, nan]]],


        [[[nan, nan, nan],
          [nan, nan, nan],
          [nan, nan, nan]]],


        [[[nan, nan, nan],
          [nan, nan, nan],
          [nan, nan, nan]]],


        ...,


        [[[nan, nan, nan],
          [nan, nan, nan],
          [nan, nan, nan]]],


        [[[nan, nan, nan],
          [nan, nan, nan],
          [nan, nan, nan]]],


        [[[nan, nan, nan],
          [nan, nan, nan],
          [nan, nan, nan]]]], device='cuda:0', requires_grad=True)

**Comment by robin1001** [comment]
please check your data.

**Comment by apoca909** [comment]
default training params
some data not works

![image](https://user-images.githubusercontent.com/4969897/125898566-8f671331-c6e9-46ca-988f-37a6adb52ade.png)

![image](https://user-images.githubusercontent.com/4969897/125898673-1712afcd-d128-4b10-b116-8fce12407020.png)


> please check your data.

**Comment by 601222543** [comment]
@apoca909 Is your problem solved?

**Comment by iou2much** [comment]
@apoca909 @601222543 
did you add use_amp option for training?

**Comment by onair1314** [comment]
same question
@apoca909 @601222543 do you have any good idea?
